### PR TITLE
argocd-task: add docs for the plugin for various operations

### PR DESCRIPTION
**Pull request by jchikatur**

### CODE REVIEW

File: docs/plugins-v2/argocd.md [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+        app: test
+        auth:
+          basic:
+            username: localUser
+            password: password
+```
+
+### LDAP Authentication
+
+In this case, Argo CD instance delegates authentication to a LDAP directory backed
```
Comment (mtkunkel) [inline docs/plugins-v2/argocd.md:131]: an LDAP

File: docs/plugins-v2/argocd.md [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+
+- `app`: Name of the application to be created. 
+- `cluster`: Name of the Kubernetes cluster on which the application is to be created.
+- `namespace`: Namespace in which the application is to be created. 
+- `createNamespace`: optional `boolean` when set will force creation of the namespace
+specified, if not present. 
+- `gitRepo`: optional SCM information to which the application will sync. Not required
+if `helmRepo` is specified.
+    * `repoURL`: URL of the SCM Repository
+    * `targetRevision`: Branchname, tag, or commitSHA to sync to.
```
Comment (mtkunkel) [inline docs/plugins-v2/argocd.md:189]: Branch name, tag or commit hash

File: docs/plugins-v2/argocd.md [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+      - name: PARAM1
+        value: value1
+      - name: PARAM2
+        value: value2
+        ...
+```
+
+### Update Application Spec
+
+Use the `updateSpec` action to updation the application resource object.
```
Comment (mtkunkel) [inline docs/plugins-v2/argocd.md:333]: updation?

**Comment by jchikatur** [comment]
Addressed the review comments

**Review by mtkunkel — APPROVED** [review]

**Pull request merged**